src/main/java/dev/morling/onebrc/CalculateAverage_royvanrijn.java

```diff
@@ -278,85 +240,69 @@ static int min(final int a, final int b) {
+        return 31 * hash + (int) (word ^ (word >>> 32));
+    }
+
+    private static long compilePattern(final byte value) {
+        return ((long) value << 56) | ((long) value << 48) | ((long) value << 40) | ((long) value << 32) |
+                ((long) value << 24) | ((long) value << 16) | ((long) value << 8) | (long) value;
+    }
+
     /**
      * A normal Java HashMap does all these safety things like boundary checks... we don't need that, we need speeeed.
      *
      * So I've written an extremely simple linear probing hashmap that should work well enough.
      */
     class MeasurementRepository {
-        private int tableSize = 1 << 20; // can grow in theory, made large enough not to (this is faster)
+        private int tableSize = 1 << 20; // large enough for the contest.
         private int tableMask = (tableSize - 1);
-        private int tableLimit = (int) (tableSize * LOAD_FACTOR);
-        private int tableFilled = 0;
-        private static final float LOAD_FACTOR = 0.8f;
 
-        private Entry[] table = new Entry[tableSize];
+        private MeasurementRepository.Entry[] table = new MeasurementRepository.Entry[tableSize];
+
+        record Entry(long address, long[] data, int length, int hash, String city, Measurement measurement) {
 
-        record Entry(int hash, long[] nameBytesInLong, String cityName, Measurement measurement) {
             @Override
             public String toString() {
-                return cityName + "=" + measurement;
+                return city + "=" + measurement;
             }
         }
 
-        public Measurement update(long[] nameBytesInLong, ByteBuffer bb, int length, int calculatedHash) {
+        public void update(long address, long[] data, int length, int hash, int temperature) {
 
-            final int nameBytesInLongLength = 1 + (length >>> 3);
-
-            int index = calculatedHash & tableMask;
-            Entry tableEntry;
+            int dataLength = length >> 3;
+            int index = hash & tableMask;
+            MeasurementRepository.Entry tableEntry;
             while ((tableEntry = table[index]) != null
-                    && (tableEntry.hash != calculatedHash || !arrayEquals(tableEntry.nameBytesInLong, nameBytesInLong, nameBytesInLongLength))) { // search for the right spot
+                    && (tableEntry.hash != hash || tableEntry.length != length || !arrayEquals(tableEntry.data, data, dataLength))) { // search for the right spot
                 index = (index + 1) & tableMask;
             }
 
             if (tableEntry != null) {
-                return tableEntry.measurement;
+                tableEntry.measurement.updateWith(temperature);
+                return;
             }
 
             // --- This is a brand new entry, insert into the hashtable and do the extra calculations (once!) do slower calculations here.
             Measurement measurement = new Measurement();
```

Comment:
Move slow path code out of the hot path, to be sure the hot will be inlined

Reply:
We need to update the measurement with the new temperature, always. I'm not sure what you mean?

Reply:
The part related the brand new entry, I didn't checked if goes beyond the inlining bytecode threshold, so If you mode it in a separate method you are sure the hot path to be inlined